modules/gdscript/gdscript_parser.cpp

```diff
@@ -4224,17 +4308,80 @@ bool GDScriptParser::export_group_annotations(const AnnotationNode *p_annotation
 
 bool GDScriptParser::warning_annotations(const AnnotationNode *p_annotation, Node *p_target, ClassNode *p_class) {
 #ifdef DEBUG_ENABLED
+	if (is_ignoring_warnings) {
+		return true; // We already ignore all warnings, let's optimize it.
+	}
+
 	bool has_error = false;
 	for (const Variant &warning_name : p_annotation->resolved_arguments) {
-		GDScriptWarning::Code warning = GDScriptWarning::get_code_from_name(String(warning_name).to_upper());
-		if (warning == GDScriptWarning::WARNING_MAX) {
+		GDScriptWarning::Code warning_code = GDScriptWarning::get_code_from_name(String(warning_name).to_upper());
+		if (warning_code == GDScriptWarning::WARNING_MAX) {
 			push_error(vformat(R"(Invalid warning name: "%s".)", warning_name), p_annotation);
 			has_error = true;
 		} else {
-			p_target->ignored_warnings.push_back(warning);
+			int start_line = p_annotation->start_line;
+			int end_line = p_target->end_line;
+
+			switch (p_target->type) {
+#define SIMPLE_CASE(m_type, m_class, m_property)          \
+	case Node::m_type: {                                  \
+		m_class *node = static_cast<m_class *>(p_target); \
+		if (node->m_property != nullptr) {                \
+			end_line = node->m_property->end_line;        \
+		} else {                                          \
+			end_line = node->start_line;                  \
+		}                                                 \
+	} break;
+
+				// Can contain properties (set/get).
+				SIMPLE_CASE(VARIABLE, VariableNode, initializer)
+
+				// Contain bodies.
+				SIMPLE_CASE(FOR, ForNode, list)
+				SIMPLE_CASE(IF, IfNode, condition)
+				SIMPLE_CASE(MATCH, MatchNode, test)
+				SIMPLE_CASE(WHILE, WhileNode, condition)
+#undef SIMPLE_CASE
+
+				case Node::CLASS: {
+					end_line = p_target->start_line;
+					for (const AnnotationNode *annotation : p_target->annotations) {
+						start_line = MIN(start_line, annotation->start_line);
+						end_line = MAX(end_line, annotation->end_line);
+					}
+				} break;
+
+					/* clang-format off */
+				/*case Node::FUNCTION: {
+					FunctionNode *function = static_cast<FunctionNode *>(p_target);
+					end_line = function->start_line;
+					for (int i = 0; i < function->parameters.size(); i++) {
+						end_line = MAX(end_line, function->parameters[i]->end_line);
+						if (function->parameters[i]->initializer != nullptr) {
+							end_line = MAX(end_line, function->parameters[i]->initializer->end_line);
+						}
+					}
+				} break;*/
+					/* clang-format on */
```

Comment:
Maybe remove that part of code and mention in a comment why it doesn't apply to functions?